pepper-apis/src/main/java/org/broadinstitute/ddp/client/GoogleBucketClient.java

```diff
@@ -49,6 +50,15 @@ public Bucket getBucket(String bucketName) {
         return storage.get(bucketName);
     }
 
+    public Bucket getBucketFromBlob(String bucketName) {
```

Comment:
So the reason we need this is that in production, the bucket that gets used is actually a Terra bucket, so there are some weird access constraints on it.  The Java API in particularly handles this badly, so this is a workaround to handle that.

Reply:
Ah okay. So if I'm understanding things correctly, I see that the `BlobListOption.prefix("")` part means to list all the top-level folders in the bucket. But I think `getBlobId().getBucket()` just returns a string, which is exactly what `bucketName` is. So I don't see what the difference is here... Is `storage.list()` doing something else important for working around access constraints? (is it caching some bucket object or something while doing the listing?)

Also, will there be a case where there's nothing in the bucket? In this case, there's nothing to iterate from the `Page`, so this will throw an exception. Is that what we're expecting?

Reply:
You're right; this doesn't make any sense.  Plus then I looked at the RGP backend code and I can't even find the code that does this.  I could have sworn I got it from there, too.  Let me switch this to use the existing getBucket method and test it against the Terra bucket and see what happens.  Maybe I was looking at an old version of the code?

Reply:
I switched it to use the existing getBucket method and everything was fine.  I must have been looking at an old version of the RGP code or something.